Mathlib/CategoryTheory/Sites/Coherent/RegularSheaves.lean

```diff
@@ -27,65 +29,197 @@ namespace CategoryTheory
+      (equalizerCondition_precomp_of_preservesPullback (e.op.inverse ⋙ P) e.functor h)⟩
+
+open WalkingParallelPair WalkingParallelPairHom in
+theorem parallelPair_pullback_initial {X B : C} (π : X ⟶ B)
+    (c : PullbackCone π π) (hc : IsLimit c) :
+    (parallelPair (C := (Sieve.ofArrows (fun (_ : Unit) => X) (fun _ => π)).arrows.categoryᵒᵖ)
+    (Y := op ((Presieve.categoryMk _ (c.fst ≫ π) ⟨_, c.fst, π, ofArrows.mk (), rfl⟩)))
+    (X := op ((Presieve.categoryMk _ π (Sieve.ofArrows_mk _ _ Unit.unit))))
+    (Quiver.Hom.op (Over.homMk c.fst))
+    (Quiver.Hom.op (Over.homMk c.snd c.condition.symm))).Initial := by
+  apply Limits.parallelPair_initial_mk
+  · intro ⟨Z⟩
+    obtain ⟨_, f, g, ⟨⟩, hh⟩ := Z.property
+    let X' : (Presieve.ofArrows (fun () ↦ X) (fun () ↦ π)).category :=
+      Presieve.categoryMk _ π (ofArrows.mk ())
+    let f' : Z.obj.left ⟶ X'.obj.left := f
+    exact ⟨(Over.homMk f').op⟩
+  · intro ⟨Z⟩ ⟨i⟩ ⟨j⟩
+    let ij := PullbackCone.IsLimit.lift hc i.left j.left (by erw [i.w, j.w]; rfl)
+    refine ⟨Quiver.Hom.op (Over.homMk ij (by simpa [ij] using i.w)), ?_, ?_⟩
+    all_goals congr
+    all_goals exact Comma.hom_ext _ _ (by erw [Over.comp_left]; simp [ij]) rfl
+
+/--
+Given a limiting pullback cone, the fork in `SingleEqualizerCondition` is limiting iff the diagram
+in `Presheaf.isSheaf_iff_isLimit_coverage` is limiting.
+-/
+noncomputable def isLimit_forkOfι_equiv (P : Cᵒᵖ ⥤ D) {X B : C} (π : X ⟶ B)
+    (c : PullbackCone π π) (hc : IsLimit c) :
+    IsLimit (Fork.ofι (P.map π.op) (equalizerCondition_w P c)) ≃
+    IsLimit (P.mapCone (Sieve.ofArrows (fun (_ : Unit) ↦ X) fun _ ↦ π).arrows.cocone.op) := by
+  let S := (Sieve.ofArrows (fun (_ : Unit) => X) (fun _ => π)).arrows
+  let X' := S.categoryMk π ⟨_, 𝟙 _, π, ofArrows.mk (), Category.id_comp _⟩
+  let P' := S.categoryMk (c.fst ≫ π) ⟨_, c.fst, π, ofArrows.mk (), rfl⟩
+  let fst : P' ⟶ X' := Over.homMk c.fst
+  let snd : P' ⟶ X' := Over.homMk c.snd c.condition.symm
+  let F : S.categoryᵒᵖ ⥤ D := S.diagram.op ⋙ P
+  let G := parallelPair (P.map c.fst.op) (P.map c.snd.op)
+  let H := parallelPair fst.op snd.op
+  have : H.Initial := parallelPair_pullback_initial π c hc
+  let i : H ⋙ F ≅ G := parallelPair.ext (Iso.refl _) (Iso.refl _) (by aesop) (by aesop)
+  refine (IsLimit.equivOfNatIsoOfIso i.symm _ _ ?_).trans (Functor.Initial.isLimitWhiskerEquiv H _)
+  refine Cones.ext (Iso.refl _) ?_
+  rintro ⟨_ | _⟩
+  all_goals aesop
+
+lemma equalizerConditionMap_iff_nonempty_isLimit (P : Cᵒᵖ ⥤ D) ⦃X B : C⦄ (π : X ⟶ B)
+    [HasPullback π π] : SingleEqualizerCondition P π ↔
+      Nonempty (IsLimit (P.mapCone
+        (Sieve.ofArrows (fun (_ : Unit) => X) (fun _ => π)).arrows.cocone.op)) := by
+  constructor
+  · intro h
+    exact ⟨isLimit_forkOfι_equiv _ _ _ (pullbackIsPullback π π) (h _ (pullbackIsPullback π π)).some⟩
+  · intro ⟨h⟩
+    exact fun c hc ↦ ⟨(isLimit_forkOfι_equiv _ _ _ hc).symm h⟩
+
+lemma equalizerCondition_iff_isSheaf (F : Cᵒᵖ ⥤ D) [Preregular C]
```

Comment:
The docstring at the beginning of the file should be changed because it refers to a lemma `isSheaf_iff_equalizerCondition`.